node/network/collator-protocol/src/validator_side/mod.rs

```diff
@@ -648,19 +588,19 @@ async fn fetch_collation(
 ) {
 	let (tx, rx) = oneshot::channel();
 
-	let PendingCollation { relay_parent, para_id, peer_id, .. } = pc;
-
-	let timeout = |collator_id, relay_parent| async move {
-		Delay::new(MAX_UNSHARED_DOWNLOAD_TIME).await;
-		(collator_id, relay_parent)
-	};
-	state
-		.collation_fetch_timeouts
-		.push(timeout(id.clone(), relay_parent.clone()).boxed());
+	let PendingCollation { relay_parent, para_id, peer_id, prospective_candidate, .. } = pc;
+	let candidate_hash = prospective_candidate.as_ref().map(ProspectiveCandidate::candidate_hash);
 
 	if let Some(peer_data) = state.peer_data.get(&peer_id) {
-		if peer_data.has_advertised(&relay_parent) {
-			request_collation(sender, state, relay_parent, para_id, peer_id, tx).await;
+		if peer_data.has_advertised(&relay_parent, candidate_hash) {
+			if request_collation(sender, state, pc, id.clone(), peer_data.version, tx).await {
```

Comment:
We need to either gate `request_collation` _or_ `Second`ing on whether the candidate has prospective depths. That condition needs to be re-evaluated when we get new active leaves or new candidate are recognized as `Seconded` in the prospective parachains subsystem for our existing leaves.

Reply:
> We should allow collators to make a few (up to MAX_DEPTH?) announcements per relay-parent and we'll keep but not act on announcements that we don't recognize with GetHypotheticalDepths. When we learn of new candidates getting seconded in the relevant fragment trees, we'd try GetHypotheticalDepths again.

From [#5054](https://github.com/paritytech/polkadot/issues/5054)

Reply:
(Deferred to #5923?)

Reply:
Yes, this code is not supposed to handle backed-parent stuff